Framework/Algorithms/test/CalcCountRateTest.h

```diff
@@ -0,0 +1,478 @@
+#ifndef MANTID_ALGORITHMS_CALC_COUNTRATE_TEST_H_
+#define MANTID_ALGORITHMS_CALC_COUNTRATE_TEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidKernel/TimeSeriesProperty.h"
+
+#include "MantidAlgorithms/CalcCountRate.h"
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/Axis.h"
+#include "MantidAPI/NumericAxis.h"
+
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+#include <numeric>
+
+using namespace Mantid;
+using Mantid::DataObjects::Workspace2D_sptr;
+using namespace Mantid::API;
+using namespace Mantid::Algorithms;
+
+class CalcCountRateTester : public CalcCountRate {
+public:
+  void setSearchRanges(DataObjects::EventWorkspace_sptr &InputWorkspace) {
+    CalcCountRate::setSourceWSandXRanges(InputWorkspace);
+  }
+  std::tuple<double, double, bool> getXRanges() const {
+    return std::tuple<double, double, bool>(m_XRangeMin, m_XRangeMax,
+                                            m_rangeExplicit);
+  }
+  void
+  setOutLogParameters(const DataObjects::EventWorkspace_sptr &InputWorkspace) {
+    CalcCountRate::setOutLogParameters(InputWorkspace);
+  }
+  void getAlgLogSettings(size_t &numLogSteps,
+                         Kernel::TimeSeriesProperty<double> const *&pNormLog) {
+    pNormLog = m_pNormalizationLog;
+    numLogSteps = m_numLogSteps;
+  }
+
+  DataObjects::EventWorkspace *getWorkingWS() { return m_workingWS.get(); }
+  void setVisWS(const std::string &wsName) {
+    this->setProperty("VisualizationWs", wsName);
+    this->checkAndInitVisWorkspace();
+  }
+  const std::vector<double> &getVisNormLog() { return m_visNorm; }
+};
+
+class CalcCountRateTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static CalcCountRateTest *createSuite() { return new CalcCountRateTest(); }
+  static void destroySuite(CalcCountRateTest *suite) { delete suite; }
+
+  void test_Init() {
+    CalcCountRate alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize())
+    TS_ASSERT(alg.isInitialized())
+  }
+  void test_ranges() {
+
+    auto sws = build_test_ws(false);
+
+    double XRangeMin, XRangeMax;
+
+    CalcCountRateTester alg;
+
+    alg.initialize();
+    alg.setProperty("Workspace", sws);
+    alg.setProperty("RangeUnits", "dSpacing");
+
+    alg.setSearchRanges(sws); // explicit ranges:
+    //-- real workspace ranges returned
+
+    auto ranget = alg.getXRanges();
```

Comment:
ranget -> ranges? Ranget might be the correct plural in Finnish, but I guess we need to stick to English here.